frontend/src/metabase/query_builder/components/view/sidebars/ChartSettingsSidebar.jsx

```diff
@@ -5,38 +5,49 @@ import ChartSettings from "metabase/visualizations/components/ChartSettings";
 import visualizations from "metabase/visualizations";
 import SidebarContent from "metabase/query_builder/components/SidebarContent";
 
-const ChartSettingsSidebar = ({
-  question,
-  result,
-  addField,
-  initialChartSetting,
-  onReplaceAllVisualizationSettings,
-  onClose,
-  onOpenChartType,
-  ...props
-}) =>
-  result && (
-    <SidebarContent
-      className="full-height"
-      title={t`${visualizations.get(question.display()).uiName} options`}
-      onDone={onClose}
-      onBack={onOpenChartType}
-    >
-      <ChartSettings
-        question={question}
-        addField={addField}
-        series={[
-          {
-            card: question.card(),
-            data: result.data,
-          },
-        ]}
-        onChange={onReplaceAllVisualizationSettings}
-        onClose={onClose}
-        noPreview
-        initial={initialChartSetting}
-      />
-    </SidebarContent>
-  );
+export default class ChartSettingsSidebar extends React.Component {
+  state = { sidebarPropsOverride: null };
 
-export default ChartSettingsSidebar;
+  setSidebarPropsOverride = sidebarPropsOverride =>
+    this.setState({ sidebarPropsOverride });
+
+  render() {
+    const {
+      question,
+      result,
+      addField,
+      initialChartSetting,
+      onReplaceAllVisualizationSettings,
+      onClose,
+      onOpenChartType,
+    } = this.props;
+    const { sidebarPropsOverride } = this.state;
+    return (
+      result && (
+        <SidebarContent
+          className="full-height"
+          title={t`${visualizations.get(question.display()).uiName} options`}
+          onDone={onClose}
+          onBack={onOpenChartType}
+          {...sidebarPropsOverride}
```

Comment:
Apparently  `{...null}` is fine! And, so is `{...undefined}`, `{...false}`, etc.

I guess that's consistent with `false.foo` not throwing an error. It just has no properties to spread, but there's nothing wrong with asking. 🤷‍♂